ee/tabby-ui/components/prompt-form.tsx

```diff
@@ -26,15 +35,56 @@ export function PromptForm({
   isLoading
 }: PromptProps) {
   const { formRef, onKeyDown } = useEnterSubmit()
-  const inputRef = React.useRef<HTMLTextAreaElement>(null)
-  const router = useRouter()
+  const [queryCompletionUrl, setQueryCompletionUrl] = React.useState<
+    string | null
+  >(null)
+  const latestFetchKey = React.useRef('')
 
-  React.useEffect(() => {
-    if (inputRef.current) {
-      inputRef.current.focus()
+  useSWR<SearchReponse>(queryCompletionUrl, fetcher, {
+    revalidateOnFocus: false,
+    dedupingInterval: 500,
+    onSuccess: (data, key) => {
+      if (key !== latestFetchKey.current) return
+
+      setOptions(data?.hits ?? [])
     }
+  })
+
+  const [options, setOptions] = React.useState<SearchReponse['hits']>([])
+  const onSearch = React.useMemo(() => {
+    return debounce((e: React.ChangeEvent<HTMLTextAreaElement>) => {
+      const value = e.target?.value ?? ''
+      const end = e.target?.selectionEnd ?? 0
+      const queryname = getSearchCompletionQueryName(value, end)
+      if (queryname) {
+        let url = `/v1beta/search?q=name:${queryname}`
+        latestFetchKey.current = url
+        setQueryCompletionUrl(`/v1beta/search?q=name:${queryname}`)
+      } else {
+        setOptions([])
+      }
+    }, 200)
   }, [])
 
+  const onSelectCompletion = (
+    inputRef: React.MutableRefObject<
+      HTMLTextAreaElement | HTMLInputElement | null
+    >,
+    item: ISearchHit
+  ) => {
+    const replaceString = '`@' + item?.doc?.name + '` '
+    const selectionEnd = inputRef.current?.selectionEnd ?? 0
+    const queryname = getSearchCompletionQueryName(input, selectionEnd)
+    const prevInput = input
+      .substring(0, selectionEnd)
+      .replace(new RegExp(`@${queryname}$`), '')
+    if (queryname) {
+      setInput(prevInput + replaceString + input.substring(selectionEnd))
+    }
+
+    setOptions([])
+  }
+
   return (
     <form
       onSubmit={async e => {
```

Comment:
When submit, @get_name needs to be replaced with `doc.body`

Reply:
Done